File: android/src/main/java/com/example/flutter_in_brain_launcher/paramaters/ConfigureInBrainParameters.java
package com.example.flutter_in_brain_launcher.paramaters;

import android.app.Activity;

import io.flutter.plugin.common.MethodCall;

public class ConfigureInBrainParameters {
    public final String apiClientId;
    public final String apiSecret;
    public final boolean isS2S;
    public final String userId;
    public final Activity activity;

    public static boolean checkIfParamsCorrect(MethodCall call) {
        return (call.hasArgument("apiClientID") && call.hasArgument("apiSecret") && call.hasArgument("userID") && call.hasArgument("isS2S"));
    }

    public ConfigureInBrainParameters(MethodCall call, Activity activity) {
        this.apiClientId = call.argument("apiClientID");
        this.apiSecret = call.argument("apiSecret");
        this.isS2S = call.argument("isS2S");
        this.userId = call.argument("userID");
        this.activity = activity;
    }
}

File: ios/Classes/Parameters/ConfigureInBrainParameters.swift
//
//  ConfigureInBrainParameters.swift
//  flutter_in_brain_launcher
//
//  Created by Илья Клюкин on 8.02.23.
//

import Foundation

struct ConfigureInBrainParameters {
    let apiClientId: String
    let  apiSecret: String
    let userId: String
    let isS2S: Bool
    
    init?(dict: Dictionary<String, Any>) {
        guard let apiCliendId = dict.unwrap("apiClientID") as? String,
              let apiSecret = dict.unwrap("apiSecret") as? String,
              let userId = dict.unwrap("userID") as? String,
              let isS2S = dict.unwrap("isS2S") as? Bool else {
            return nil
        }
        self.apiClientId = apiCliendId
        self.apiSecret = apiSecret
        self.userId = userId
        self.isS2S = isS2S
    }
}
